collections/src/main/scala/strawman/collection/immutable/ImmutableArray.scala

```diff
@@ -124,36 +129,34 @@ sealed abstract class ImmutableArray[+A]
   * @define coll immutable array
   * @define Coll `ImmutableArray`
   */
-object ImmutableArray extends StrictOptimizedSeqFactory[ImmutableArray] {
-
-  private[this] lazy val emptyImpl = new ImmutableArray.ofRef[Nothing](new scala.Array[Nothing](0))
+object ImmutableArray extends StrictOptimizedClassTagSeqFactory[ImmutableArray] { self =>
+  val untagged: SeqFactory[ImmutableArray] = new ClassTagSeqFactory.AnySeqDelegate(self)
 
-  def empty[A]: ImmutableArray[A] = emptyImpl
+  private[this] lazy val emptyImpl = new ImmutableArray.ofRef[Nothing](new Array[Nothing](0))
```

Comment:
Why does this need to be `lazy`?

Reply:
It doesn't need to be but there is usually no reason to ever allocate these objects.